src/Bicep.Core/Parsing/Parser.cs

```diff
@@ -51,6 +51,23 @@ public ProgramSyntax Program()
                         declarationsOrTokens.Add(newLine);
                     }
                 }
+
+                if (this.IsAtEnd() &&
+                    declarationOrToken is MissingDeclarationSyntax missingDeclaration &&
+                    !missingDeclaration.HasParseErrors())
+                {
+                    // If there are dangling decorators and we hit EOF, ask users to add a declration.
+                    // Set the span of the diagnostic so that it's on the line below the last decorator.
+                    var lastNewLine = missingDeclaration.LeadingNodes.Last(node => node is Token { Type: TokenType.NewLine });
+                    var diagnosticSpan = new TextSpan(lastNewLine.Span.Position + 2, 0);

```

Comment:
The `+ 2` approach only works on CRLF files. On LF, the position is off by 1:

![image](https://user-images.githubusercontent.com/22460039/106188051-51f71e80-615b-11eb-8f9c-3701f569862f.png)

If you don't have a newline following the decorator, the parser will tell you to add it, so it seems like you can rely on it and use its position, but not sure if there are other edge cases.

Reply:
I'm fine if this goes into the next PR also.

---
In reply to: [566350593](https://github.com/Azure/bicep/pull/1324#discussion_r566350593) [](ancestors = 566350593)

Reply:
This is to handle an edge case where there are multiple newlines after a decorator:

![image](https://user-images.githubusercontent.com/16367959/106189075-a8b12800-615c-11eb-9133-6b1ac80d34fe.png)


Not sure why the error span is [3, 2]. On my machine it's [3, 1]...

Reply:
It's due to different line endings.